diode-react/src/main/scala/diode/react/ReactConnector.scala

```diff
@@ -115,22 +116,24 @@ trait ReactConnector[M <: AnyRef] { circuit: Circuit[M] =>
       }
 
       private def changeHandler(cursor: ModelRO[S]): Unit = {
-        // modify state if we are mounted and state has actually changed
-        if (t.isMounted() && modelReader =!= t.accessDirect.state) {
-          t.accessDirect.setState(modelReader())
-        }
+        val isMounted = t.isMounted
+        val stateHasChanged = t.state.map(state => modelReader =!= state)
+        (isMounted && stateHasChanged).map {
+          case true => t.setState(modelReader()).runNow()
+          case _ =>
+        }.runNow()
       }
 
-      def render(s: S, compB: ModelProxy[S] => ReactElement) = wrap(modelReader)(compB)
+      def render(s: S, compB: ModelProxy[S] => VdomElement) = wrap(modelReader)(compB)
     }
 
-    ReactComponentB[ModelProxy[S] => ReactElement]("DiodeWrapper")
+    ScalaComponent.build[ModelProxy[S] => VdomElement]("DiodeWrapper")
       .initialState(modelReader())
       .renderBackend[Backend]
-      .componentWillMount(scope => scope.backend.willMount)
-      .componentWillUnmount(scope => scope.backend.willUnmount)
-      .shouldComponentUpdate(scope => (scope.currentState ne scope.nextState) || (scope.currentProps ne scope.nextProps))
+      .componentWillMount(_.backend.willMount)
+      .componentWillUnmount(_.backend.willUnmount)
+      .shouldComponentUpdatePure(scope => (scope.currentState ne scope.nextState) || (scope.currentProps ne scope.nextProps))
       .build
-      .set(key)
+      .withRawProp("key", key)
```

Comment:
The new API also offers `withKey(k: Key)`, so maybe you want to change the public API of `connect` to use a `react.Key`?